scripts/install_vim.sh

```diff
@@ -1,90 +1,220 @@
-    apk add python3-dev
+  if [ -n "$PYTHON3" ]; then
+    apk add --virtual vim-build python3-dev
+    if [ "$FLAVOR" = vim ]; then
+      CONFIG_ARGS="$CONFIG_ARGS --enable-python3interp=dynamic"
+    else
+      apk add python3
+      pip3 install neovim
+    fi
   fi
 
   if [ $RUBY -eq 1 ]; then
-    CONFIG_ARGS="$CONFIG_ARGS --enable-rubyinterp"
-    apk add ruby-dev
+    apk add --virtual vim-build ruby-dev
+    apk add ruby
+    if [ "$FLAVOR" = vim ]; then
+      CONFIG_ARGS="$CONFIG_ARGS --enable-rubyinterp"
+    else
+      apk add --virtual vim-build ruby-rdoc ruby-irb
+      gem install neovim
+    fi
   fi
 
   if [ $LUA -eq 1 ]; then
-    CONFIG_ARGS="$CONFIG_ARGS --enable-luainterp"
-    apk add lua-dev
+    if [ "$FLAVOR" = vim ]; then
+      CONFIG_ARGS="$CONFIG_ARGS --enable-luainterp"
+      apk add --virtual vim-build lua-dev
+      apk add lua
+    else
+      echo 'WARNING: -lua is not supported with Neovim, ignoring.'
+    fi
   fi
 
-  if [ -n "$CONFIGURE_OPTIONS" ]; then
+  if [ "$FLAVOR" = vim ] && [ -n "$CONFIGURE_OPTIONS" ]; then
     CONFIG_ARGS="$CONFIG_ARGS $CONFIGURE_OPTIONS"
   fi
 
   cd /vim
 
-  BUILD_DIR="vim-${TAG#v}"
-
-  apk info -q vim-build > /dev/null \
-    || apk add --virtual vim-build make ncurses-dev curl gcc libc-dev
+  if [ -e "$INSTALL_PREFIX" ]; then
+    echo "WARNING: $INSTALL_PREFIX exists already.  Overwriting."
+  fi
 
-  if [ ! -d $BUILD_DIR ]; then
+  BUILD_DIR="${FLAVOR}-${repo}-${tag}"
+  if [ ! -d "$BUILD_DIR" ]; then
+    mkdir -p "$BUILD_DIR"
+    cd "$BUILD_DIR"
     # The git package adds about 200MB+ to the image.  So, no cloning.
-    echo "Downloading $TAG"
-    curl -SL "https://github.com/vim/vim/archive/${TAG}.tar.gz" | tar zx
+    echo "Downloading $repo:$tag"
+    curl -SL "https://github.com/$repo/archive/${tag}.tar.gz" | tar zx --strip-components=1
+  else
+    cd "$BUILD_DIR"
   fi
 
-  cd $BUILD_DIR
-  echo "Configuring with: $CONFIG_ARGS"
-  # shellcheck disable=SC2086
-  ./configure $CONFIG_ARGS || bail "Could not configure"
-  make CFLAGS="-U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2" -j4 || bail "Make failed"
-  make install || bail "Install failed"
+  if [ "$FLAVOR" = vim ]; then
+    apk add --virtual vim-build ncurses-dev
+    apk add ncurses
+
+    echo "Configuring with: $CONFIG_ARGS"
+    # shellcheck disable=SC2086
+    ./configure $CONFIG_ARGS || bail "Could not configure"
+    make CFLAGS="-U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2" -j4 || bail "Make failed"
+    make install || bail "Install failed"
 
-  ln -s $VIM_BIN /vim-build/bin/$NAME
+  else
+    apk add libuv \
+      libtermkey \
+      libvterm \
+      msgpack-c \
+      unibilium
+    apk add --virtual vim-build \
+      autoconf \
+      automake \
+      ca-certificates \
+      cmake \
+      g++ \
+      gperf \
+      libtool \
+      libuv-dev \
+      libtermkey-dev \
+      libvterm-dev \
+      luajit \
+      luajit-dev \
+      m4 \
+      make \
```

Comment:
Oh, I was referring to things like `make`.  It's not installed by the vim flavor, but it still uses `make`, right?